### PR TITLE
Introduce StickyHeaderView in UIScrollView

### PR DESCRIPTION
**Pull request by muukii**
CleanShot 2022-03-30 at 00 02 41

### CODE REVIEW

File: ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    target.updateContentInset()
+  }
+}
+
+/**
+ With: ``ScrollStickyContentType``
+ */
+public final class ScrollStickyVerticalHeaderView: UIView {
```
Comment (muukii) [inline ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift:21]: New

File: ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift [hunk 2]
```diff
@@ -0,0 +1,165 @@
+import UIKit
+
+public protocol ScrollStickyContentType: UIView {
```
Comment (muukii) [inline ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift:3]: And new

File: ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    self.observations = newObservations
+  }
+ 
+  func updateContentInset() {
+
+    guard let targetScrollView = targetScrollView else {
+      return
+    }
+
+    targetScrollView.contentInset.top = 0
```
Comment (takumatt) [inline ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift:143]: Is it safe to set it to zero here?

**Review by takumatt — APPROVED** [review]

File: ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift (hunk 3 above)
Comment (muukii) [reply ScrollEdgeControl/Core/ScrollStickyVerticalHeaderView.swift:143]: for now, to get default adjustment content inset according to only safe-area-insets by setting 0 before setting new content inset.

**Pull request merged**